config/fs.go

```diff
@@ -0,0 +1,239 @@
+
+	// NegativeTimeout defines the overall entry timeout for failed lookups.
+	NegativeTimeout int64 `toml:"negative_timeout"`
+
+	// LogFuseOperations enables logging of operations on FUSE FS. This is to be used
+	// for debugging purposes only. This option may emit sensitive information,
+	// e.g. filenames and paths within an image
+	LogFuseOperations bool `toml:"log_fuse_operations"`
+}
+
+type BackgroundFetchConfig struct {
+	Disable bool `toml:"disable"`
+
+	// SilencePeriodMsec defines the time (in ms) the background fetcher
+	// will be paused for when a new image is mounted.
+	SilencePeriodMsec int64 `toml:"silence_period_msec"`
+
+	// FetchPeriodMsec specifies how often a background fetch will occur.
+	// The background fetcher will fetch one span every FetchPeriodMsec.
+	FetchPeriodMsec int64 `toml:"fetch_period_msec"`
+
+	// MaxQueueSize specifies the maximum size of the work queue
+	// i.e., the maximum number of span managers that can be queued
+	// in the background fetcher.
+	MaxQueueSize int `toml:"max_queue_size"`
+
+	// EmitMetricPeriodSec is the amount of interval (in second) at which the background
+	// fetcher emits metrics
+	EmitMetricPeriodSec int64 `toml:"emit_metric_period_sec"`
+}
+
+// RetryConfig represents the settings for retries in a retryable http client.
+type RetryConfig struct {
+	// MaxRetries is the maximum number of retries before giving up on a retryable request.
+	// This does not include the initial request so the total number of attempts will be MaxRetries + 1.
+	MaxRetries int
+	// MinWait is the minimum wait time between attempts. The actual wait time is governed by the BackoffStrategy,
+	// but the wait time will never be shorter than this duration.
+	MinWaitMsec int64
+	// MaxWait is the maximum wait time between attempts. The actual wait time is governed by the BackoffStrategy,
+	// but the wait time will never be longer than this duration.
+	MaxWaitMsec int64
+}
+
+// TimeoutConfig represents the settings for timeout at various points in a request lifecycle in a retryable http client.
+type TimeoutConfig struct {
+	// DialTimeout is the maximum duration that connection can take before a request attempt is timed out.
+	DialTimeoutMsec int64
+	// ResponseHeaderTimeout is the maximum duration waiting for response headers before a request attempt is timed out.
+	// This starts after the entire request body is uploaded to the remote endpoint and stops when the request headers
+	// are fully read. It does not include reading the body.
+	ResponseHeaderTimeoutMsec int64
+	// RequestTimeout is the maximum duration before the entire request attempt is timed out. This starts when the
+	// client starts the connection attempt and ends when the entire response body is read.
+	RequestTimeoutMsec int64
+}
+
+// RetryableHTTPClientConfig is the complete config for a retryable http client
+type RetryableHTTPClientConfig struct {
+	TimeoutConfig
+	RetryConfig
+}
+
+func parseFSConfig(cfg *Config) {
```

Comment:
Instead of this monolithic function that constructs all the nested structs, can we have separate reusable constructors for each struct and call them from here?